flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -49,6 +55,28 @@
 	// Actor which submits a job to the JobManager via this actor
 	private ActorRef submitter;
 
+	// timeout for a message from the job manager
+	private static FiniteDuration JOB_CLIENT_JOB_MANAGER_TIMEOUT = new FiniteDuration(10000, TimeUnit.MILLISECONDS);
+
+	// heartbeat interval for pinging the job manager for job status
+	private static FiniteDuration JOB_CLIENT_HEARTBEAT_INTERVAL = new FiniteDuration(5000, TimeUnit.MILLISECONDS);
+
+	// initial time delay before starting pinging job manager over regular intervals
+	private static FiniteDuration JOB_CLIENT_INITIAL_PING_DELAY = new FiniteDuration(500, TimeUnit.MILLISECONDS);
+
+	// maximum waiting time for a job to go to running status (milliseconds)
+	// setting highly conservative limit for now. This should ideally be in Configuration.
+	private static long JOB_CLIENT_JOB_STATUS_TIMEOUT = 30000;
```

Comment:
These values should be configurable. Thus they should be definable in the `flink-conf.yaml` file.